tests/integration/test_charm.py

```diff
@@ -37,19 +37,25 @@ async def test_build_and_deploy(ops_test: OpsTest):
         raise_on_blocked=True,
         timeout=1000,
     )
+    assert len(ops_test.model.applications[APP_NAME].units) == 2
     assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
+    assert ops_test.model.applications[APP_NAME].units[1].workload_status == "active"
 
     # effectively disable the update status from firing
     await ops_test.model.set_config({"update-status-hook-interval": "60m"})
 
 
 @pytest.mark.abort_on_fail
-async def test_application_is_up(ops_test: OpsTest):
-    status = await ops_test.model.get_status()  # noqa: F821
-    address = status["applications"][APP_NAME]["units"][f"{APP_NAME}/0"]["address"]
-
-    url = f"http://{address}"
-
-    logger.info("querying app address: %s", url)
-    response = urllib.request.urlopen(url, data=None, timeout=2.0)
-    assert response.code == 200
+async def test_database_package_installation(ops_test: OpsTest):
+    """Confirm that the charm units contain installed software."""
+    # Test each MySQL unit
+    for unit in ops_test.model.applications[APP_NAME].units:
+        # Ensure that mysql-server is installed correctly
+        result = await run_command_on_unit(unit, "mysqld --version")
+        mysql_version = result.strip().split()
+        assert mysql_version[2] == "8.0.28-0ubuntu0.20.04.3"
```

Comment:
Would test the command return code (e.g. == 0) a more robust method? 
The concern is that, when we have a new package release, the test will fail, even though the functionality wont.

Reply:
my concern was that packages would change (since we have not yet determined how to version lock, and minor versions can have breaking changes) and so wanted to test the exact version to alert us if this happens. thoughts on keeping as is and revisiting once we have looked into version locking?

Reply:
@delgod What's your experience with mysql stuff, concerning version compatibility breaks?

@shayancanonical eventually, testing `major.minor` could be a nice middle ground.